src/QsCompiler/CompilationManager/CompilationUnit.cs

```diff
@@ -112,8 +112,10 @@ static QsDeclarationAttribute Renamed(QsQualifiedName originalName, Position dec
                 var attArg = SyntaxGenerator.StringLiteral(
                     originalName.ToString(),
                     ImmutableArray<TypedExpression>.Empty);
+
```

Comment:
The deprecation warning not showing up might be an interop bug. There are also two related problems within our control:

1. Our record implementations are public, meaning that records have a single autogenerated constructor that requires all of their fields (even deprecated fields).
2. The alternative record constructors are defined in F# as optional type extensions, meaning they are inaccessible from C#, so C# code currently has no way to construct a UserDefinedType that won't break when the range field is removed.

> I think it might require to not use records as part of the AST unfortunately - do you have other ideas and would be more pragmatic given that we do use records currently?

Records are fine - it's specifically the public fields that are problematic for backwards compatibility.

Reply:
Yes, agreed. I am also wondering if it might be worthwhile to consider moving the constructor extensions to the same file as the declaration to make them available in C#.

Reply:
Made issue #958.